src/main/java/sirius/db/redis/Redis.java

```diff
@@ -219,7 +219,7 @@ private Jedis getConnection() {
         return setupConnection();
     }
 
-    private Jedis setupConnection() {
+    private synchronized Jedis setupConnection() {
```

Comment:
add an else if (sentinelPool != null) {
return sentinelPool.getResource();
} in line 230 - to be 100% race-condition free